Tasks/AzureRmWebAppDeployment/Utility.ps1

```diff
@@ -2,13 +2,33 @@ $ErrorActionPreference = 'Stop'
 
 function Get-MsDeployExePath
 {
-    $currentDir = (Get-Item -Path ".\").FullName
+    $MSDeployExePath = $null
+    try
+    {
+        $MSDeployExePath , $MSDeployVersion = Get-MSDeployOnTargetMachine
+    }
+    catch [System.Exception]
+    {
+         Write-Verbose ("MSDeploy is not installed in system." + $_.Exception.Message)
+    }
 
-    $msDeployExeDir = Join-Path $currentDir "MSDeploy3.6"
-    $msDeployExePath = Join-Path $msDeployExeDir "msdeploy.exe"
-    Write-Host (Get-LocalizedString -Key "msdeploy.exe is located at '{0}'" -ArgumentList $msDeployExePath)
+    if( $MSDeployExePath -ne $null -and $MSDeployVersion -lt 3 ){
+        throw  "Unsupported installed version found for MSDeploy"
```

Comment:
Can you also call out the supported version. User should know what should be corrective action.